### PR TITLE
add support for modern OSX

### LINKED ISSUE #20: mavericks
on newer OSX (currently mavericks and yosemite) versions, the `pkgType` is `mac.binary.mavericks`, and the path `file.path("bin", "macosx", "mavericks", "contrib", rversion)`

[you don’t handle that](https://github.com/eddelbuettel/drat/blob/master/R/insertPackage.R#L164)

### PR DESCRIPTION
**Pull request by flying-sheep**
fixes #20. maybe.

### CODE REVIEW

File: R/insertPackage.R [hunk 1]
```diff
@@ -130,7 +130,7 @@ identifyPackageType <- function(file) {
     ret <- if (grepl("_.*\\.tar\\..*$", file)) {
         "source"
     } else if (grepl("_.*\\.tgz$", file)) {
-        "mac.binary"
+        "mac.binary.mavericks"
```
Comment (eddelbuettel) [inline R/insertPackage.R]: That doesn't look right.  As long as Simon supports both flavors, shouldn't we too?

Comment (JanSchulz) [inline R/insertPackage.R]: This will break later, as `write_PACKAGES` only accepts `"mac.binary"` :-(

Comment (flying-sheep) [inline R/insertPackage.R]: @JanSchulz did you report that bug on the R bug tracker?

Comment (eddelbuettel) [inline R/insertPackage.R]: Maybe it is the defined behaviour?   Windows has only one binary type...

Comment (flying-sheep) [inline R/insertPackage.R]: certainly not. miniCRAN also [hacks around it](https://github.com/RevolutionAnalytics/miniCRAN/blob/master/R/makeRepo.R#L85)

Comment (eddelbuettel) [inline R/insertPackage.R]: That is exactly what I said: as far as the PACKAGES format goes, there is one type for OS X and one type for Windows.

**Comment by flying-sheep** [comment]
you can still merge the first commit, and we’ll have to come up with a way to look into the .tgz files to find out what type they are (instead of the second commit)

**Comment by eddelbuettel** [comment]
Yes, will merge proper OS X support but this did not look right to @JanSchulz or myself.  Let's rework, and preferably _test_ on two or more OS X flavours.

**Comment by flying-sheep** [comment]
well, the first commit is trivially correct, and the second one makes us incompatible with legacy OSX versions while making us compatible with current ones.

i’m rather incompatible with old versions than being *only* complatible with old versions.

**Comment by flying-sheep** [comment]
i don’t even really care about OSX. as long as i don’t get to run a free version of something in a VM inside of linux, it’s not worth supporting more than half-assedly.

**Pull request closed without merging**